app/services/hyku_addons/collection_branding_fetcher.rb

```diff
@@ -2,7 +2,8 @@
 
 module HykuAddons
   class CollectionBrandingFetcher
-    def self.collection_banner(id)
+    def self.collection_banner(id, cname)
+      AccountElevator.switch!(cname)
       @_collection_banner ||= collection_branding_info(id).where(role: 'banner')
```

Comment:
The class instance variable here i going to cause issues. Essentially, once set, it will not be unset until the server restarts as that is when the class is reloaded. 

```
2.7.4 :001 > class Foo
2.7.4 :002 >   def self.bar(number)
2.7.4 :003 >     @number ||= number
2.7.4 :004 >   end
2.7.4 :005 > end
 => :bar
2.7.4 :006 > foo = Foo.new
 => #<Foo:0x0000565351b6da28>
2.7.4 :007 > Foo.bar(1234)
 => 1234
2.7.4 :008 > Foo.bar(5678)
 => 1234
2.7.4 :009 > bar = Foo.new
 => #<Foo:0x0000565350282f40>
2.7.4 :010 > foo.class.instance_variable_get(:@number)
 => 1234
2.7.4 :011 > bar.class.instance_variable_get(:@number)
 => 1234
```
As you can see from this example, when set, all other instances of the the class will use whatever if the first value set, and all other times the class method is called will return the first value set. 

If you want to use a class method to set the banner, you cannot use a memoized variable inside of it.

Reply:
Thanks @prdanelli, the code has been updated to use instance methods.